shared/src/main/scala/mlscript/helpers.scala

```diff
@@ -715,6 +715,14 @@ trait LitImpl { self: Lit =>
 }
 
 trait VarImpl { self: Var =>
+  /** Check if the variable name is an integer. */
+  def isIndex: Bool = name.headOption match {
+    case S('0') => name.length === 1
+    case S(_) => name.forall(_.isDigit)
+    case N => false
+  }
+  /** Get the integer if it's a valid index. */
+  def toIndexOption: Opt[Int] = if (isIndex) name.toIntOption else N
```

Comment:
Why duplicate the implementation? This is already defined in `helpers`.

Reply:
The purpose of these functions is different. `isIndex` is used to determine whether it is a valid tuple index. On the other hand, `toIndexOption` performs the check and returns this index as `Int`.

With `toIndexOption`, we can simplify related code in `TypeSimplifier` and `TyperHelpers` to one-liners.

Reply:
Never mind, I misread it!